conf/broker.conf

```diff
@@ -85,6 +85,7 @@ advertisedAddress=
 # internalListenerName=
 
 # Enable or disable the HAProxy protocol.
+# If true, the real IP addresses of consumers and producers can be obtained when getting topic statistics data.
```

Comment:
Real might be misleading here. I think the IP address is the remote IP address from the proxy's perspective? If you agree, I think we should clarify further to prevent confusion. This is relevant when traffic ingresses to the proxy through any kind of NAT. A good example is a kubernetes load balancer with `externalTrafficPolicy: cluster`.

Reply:
I also found this setting very misleading the first time I encountered it. We're not really even using the HA Proxy protocol. It seems that we appropriated the protocol message to propagate the client IP address, which seems worth documenting since the HAProxy part is largely irrelevant here.

Reply:
It looks like I might be somewhat mistaken. This PR has some relevant details https://github.com/apache/pulsar/pull/16045. The edge case is that you can get the "wrong" address if the inbound connection to the pulsar proxy does not start with the HAProxy message. Here is the relevant code:

https://github.com/apache/pulsar/blob/15453964f6e6086b0c5e34736958cf749f16f5e1/pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/DirectProxyHandler.java#L240-L261